#### Brief Summary

The differences in responses of the two groups regarding perspectives on rideshare and self-driving vehicles do not seem statistically significant, showing that the groups have similar beliefs on these topics prior to participating in the experiment.

In [3]:
import pandas as pd
import numpy as np
import scipy
from scipy.stats import mannwhitneyu, kruskal
import re

Loading the initial data

In [4]:
df = pd.read_csv("preride_data.csv")

In [5]:
df.shape

(22, 39)

In [6]:
df

Timestamp System  \
0             NaN  Julie   
1   7/27/18 10:41   Lily   
2   7/31/18 13:08  Julie   
3    8/3/18 10:46  Julie   
4    8/3/18 12:09   Lily   
5    8/6/18 14:37  Julie   
6    8/7/18 11:48   Lily   
7    8/7/18 13:06  Julie   
8    8/7/18 14:41   Lily   
9   8/10/18 10:50  Julie   
10  8/10/18 12:19   Lily   
11  8/13/18 13:09  Julie   
12  8/13/18 14:50   Lily   
13  8/14/18 11:42   Lily   
14  8/20/18 13:17   Lily   
15  8/20/18 14:50  Julie   
16  8/21/18 11:55   Lily   
17  8/21/18 13:12  Julie   
18  8/24/18 10:45  Julie   
19  8/24/18 12:14   Lily   
20  8/27/18 13:14  Julie   
21  8/27/18 14:36   Lily   

   Do you get motion sickness when looking at visual media (e.g., emails, videos, etc.) in cars/vehicles?  \
0                                                  No                                                       
1                                                  No                                                       
2                                                  No                                                       
3                                           Sometimes                                                       
4                                                  No                                                       
5                                                  No                                                       
6                                           Sometimes                                                       
7                                              rarely                                                       
8                                           Sometimes                                                       
9                                                  No                                                       
10                                        Very rarely                                                       
11                                                 No                                                       
12                                          Sometimes                                                       
13                                          Sometimes                                                       
14                                                 No                                                       
15                                                 No                                                       
16                                          Sometimes                                                       
17                                          Sometimes                                                       
18                                                 No                                                       
19                                                 No                                                       
20                                          Sometimes                                                       
21                                          Sometimes                                                       

   If you need to go somewhere by car, which would you prefer?  \
0                                        Drive myself            
1                Ask friend or family member to drive            
2                                        Drive myself            
3                                        Drive myself            
4                                        Drive myself            
5                Ask friend or family member to drive            
6                                        Drive myself            
7                                        Drive myself            
8                                   Take bus or train            
9                                        Drive myself            
10                                       Drive myself            
11                                       Drive myself          

In [7]:
# remove timestamp
df = df[df.columns[1:]]
# remove names
df = df.drop(df.columns[33],axis=1)

In [8]:
df.shape

(22, 37)

In [9]:
pd.DataFrame(list(df.columns.values))

0
0                                              System
1   Do you get motion sickness when looking at vis...
2   If you need to go somewhere by car, which woul...
3   Do you use a GPS/geography-based app? If so, w...
4   Please choose the response that best represent...
5   Please choose the response that best represent...
6   Please choose the response that best represent...
7   Please choose the response that best represent...
8   Have you ridden in a semi-autonomous or self-d...
9   If you have ridden in a semi-autonomous or sel...
10  If you have ridden in a semi-autonomous or sel...
11  Please explain why you have chosen the above r...
12  Provide 5 words (adjectives) to characterize/d...
13  Please choose the response that best represent...
14  Please choose the response that best represent...
15  Please choose the response that best represent...
16  Please choose the response that best represent...
17  Please choose the response that best represent...
18  Please choose the response that best represent...
19  Please choose the response that best represent...
20  Please choose the response that best represent...
21  Please choose the response that best represent...
22  Please choose the response that best represent...
23  Please choose the response that best represent...
24  Please choose the response that best represent...
25  Please choose the response that best represent...
26  Please choose the response that best represent...
27  Please choose the response that best represent...
28  Please choose the response that best represent...
29  Please choose the response that best represent...
30  Please choose the response that best represent...
31  Please choose the response that best represent...
32  Please choose the response that best represent...
33                                                Age
34                                             Gender
35                                         Occupation
36  Highest level of education you have completed ...

In [10]:
group_by_system = df.groupby(['System'])

An equal number of participants have been assigned to Julie and Lily

In [11]:
group_by_system.size()

System
Julie    11
Lily     11
dtype: int64

In [12]:
df.head()

System  \
0  Julie   
1   Lily   
2  Julie   
3  Julie   
4   Lily   

  Do you get motion sickness when looking at visual media (e.g., emails, videos, etc.) in cars/vehicles?  \
0                                                 No                                                       
1                                                 No                                                       
2                                                 No                                                       
3                                          Sometimes                                                       
4                                                 No                                                       

  If you need to go somewhere by car, which would you prefer?  \
0                                       Drive myself            
1               Ask friend or family member to drive            
2                                       Drive myself            
3                                       Drive myself            
4                                       Drive myself            

  Do you use a GPS/geography-based app? If so, which do you use?  \
0                                        Google Maps               
1                                        Google Maps               
2                                        Google Maps               
3                                         Apple Maps               
4                                        Google Maps               

  Please choose the response that best represents your view on taking a Lyft or Uber. [I am usually comfortable  being driven.]  \
0                                              Agree                                                                              
1                                     Strongly agree                                                                              
2                                              Agree                                                                              
3                                     Strongly agree                                                                              
4                                              Agree                                                                              

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually feel safe being driven.]  \
0                                              Agree                                                                        
1                                              Agree                                                                        
2                                              Agree                                                                        
3                                              Agree                                                                        
4                                              Agree                                                                        

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust the driver.]  \
0                                              Agree                                                                  
1                                              Agree                                                                  
2                                              Agree                                                                  
3                                              Agree                                                                  
4                                              Agree                                                                  

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust other drivers on the road.]  \
0                                            Neutral                                         

## Perspectives on rideshare (Lyft/Uber): 4 questions

Here we get only the questions that deal with rideshare and assign the Likert scale responses numerical values as follows:
'Strongly disagree':1,'Disagree':2,'Neutral':3,'Agree':4,'Strongly agree':5

In [13]:
lyft_uber_views = df[df.columns[:8]]
lyft_uber_views = lyft_uber_views.drop(lyft_uber_views.columns[1:4],axis=1)
lyft_uber_views = lyft_uber_views.replace({'Strongly disagree':1,'Disagree':2,'Neutral':3,'Agree':4,'Strongly agree':5})
lyft_uber_views

System  \
0   Julie   
1    Lily   
2   Julie   
3   Julie   
4    Lily   
5   Julie   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
10   Lily   
11  Julie   
12   Lily   
13   Lily   
14   Lily   
15  Julie   
16   Lily   
17  Julie   
18  Julie   
19   Lily   
20  Julie   
21   Lily   

    Please choose the response that best represents your view on taking a Lyft or Uber. [I am usually comfortable  being driven.]  \
0                                                   4                                                                               
1                                                   5                                                                               
2                                                   4                                                                               
3                                                   5                                                                               
4                                                   4                                                                               
5                                                   4                                                                               
6                                                   4                                                                               
7                                                   4                                                                               
8                                                   4                                                                               
9                                                   4                                                                               
10                                                  5                                                                               
11                                                  4                                                                               
12                                                  3                                                                               
13                                                  4                                                                               
14                                                  3                                                                               
15                                                  5                                                                               
16                                                  4                                                                               
17                                                  5                                                                               
18                                                  5                                                                               
19                                                  5                                                                               
20                                                  4                                                                               
21                                                  5                                                                               

    Please choose the response that best represents your view on taking a Lyft or Uber. [I usually feel safe being driven.]  \
0                                                   4                                                                         
1                                                   4                                                                         
2                                                   4                                                                         
3                                                   4                                                                         
4       

Getting the average response per group for rideshare questions

In [14]:
lyft_uber_views_group_by_system = lyft_uber_views.groupby(['System'])
lyft_uber_views_group_by_system.mean()

Please choose the response that best represents your view on taking a Lyft or Uber. [I am usually comfortable  being driven.]  \
System                                                                                                                                  
Julie                                            4.363636                                                                               
Lily                                             4.181818                                                                               

        Please choose the response that best represents your view on taking a Lyft or Uber. [I usually feel safe being driven.]  \
System                                                                                                                            
Julie                                            4.181818                                                                         
Lily                                             4.090909                                                                         

        Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust the driver.]  \
System                                                                                                                      
Julie                                            4.181818                                                                   
Lily                                             3.818182                                                                   

        Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust other drivers on the road.]  
System                                                                                                                                    
Julie                                            3.272727                                                                                 
Lily                                             2.909091

In [15]:
lyft_uber_views[lyft_uber_views.columns[1:5]].aggregate(axis = 1, func = sum)

0     15
1     15
2     14
3     17
4     15
5     15
6     16
7     14
8     13
9     14
10    18
11    16
12    11
13    14
14    12
15    18
16    15
17    19
18    18
19    17
20    16
21    19
dtype: int64

In [16]:
df_rideshare = df.assign(Rideshare = lyft_uber_views[lyft_uber_views.columns[1:5]].aggregate(axis = 1, func = sum) / 4)

In [17]:
df_rideshare.head()

System  \
0  Julie   
1   Lily   
2  Julie   
3  Julie   
4   Lily   

  Do you get motion sickness when looking at visual media (e.g., emails, videos, etc.) in cars/vehicles?  \
0                                                 No                                                       
1                                                 No                                                       
2                                                 No                                                       
3                                          Sometimes                                                       
4                                                 No                                                       

  If you need to go somewhere by car, which would you prefer?  \
0                                       Drive myself            
1               Ask friend or family member to drive            
2                                       Drive myself            
3                                       Drive myself            
4                                       Drive myself            

  Do you use a GPS/geography-based app? If so, which do you use?  \
0                                        Google Maps               
1                                        Google Maps               
2                                        Google Maps               
3                                         Apple Maps               
4                                        Google Maps               

  Please choose the response that best represents your view on taking a Lyft or Uber. [I am usually comfortable  being driven.]  \
0                                              Agree                                                                              
1                                     Strongly agree                                                                              
2                                              Agree                                                                              
3                                     Strongly agree                                                                              
4                                              Agree                                                                              

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually feel safe being driven.]  \
0                                              Agree                                                                        
1                                              Agree                                                                        
2                                              Agree                                                                        
3                                              Agree                                                                        
4                                              Agree                                                                        

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust the driver.]  \
0                                              Agree                                                                  
1                                              Agree                                                                  
2                                              Agree                                                                  
3                                              Agree                                                                  
4                                              Agree                                                                  

  Please choose the response that best represents your view on taking a Lyft or Uber. [I usually trust other drivers on the road.]  \
0                                            Neutral                                         

## Mann-Whitney U Test: Perspectives on rideshare

In [18]:
print(list(df_rideshare)[-1])

Rideshare


In [19]:
x = df_rideshare.loc[df_rideshare['System']=='Julie'][list(df_rideshare)[-1]]
y = df_rideshare.loc[df_rideshare['System']=='Lily'][list(df_rideshare)[-1]]
mannwhitneyu(x, y, alternative = 'two-sided')

MannwhitneyuResult(statistic=74.5, pvalue=0.36989955696598253)

Assuming $\alpha$ = 0.05, and given the resulting `p-value = 0.36989955696598253` we see there does not seem to be a significant difference between the two groups' perspectives on rideshare

## Demographics (age, gender, occupation, level of education)

In [20]:
demographics = df.drop(df.columns[1:-4],axis=1)
demographics.head()

System  Age  Gender        Occupation  \
0  Julie   54    Male           manager   
1   Lily   54    Male            writer   
2  Julie   26    Male  Graduate Student   
3  Julie   29  Female       PhD student   
4   Lily   33  Female          Software   

  Highest level of education you have completed or currently pursuing  
0                                  Bachelor's degree                   
1                                  Bachelor's degree                   
2                                    Doctoral degree                   
3                                    Doctoral degree                   
4                                  Bachelor's degree

In [21]:
demographics.groupby(['Gender']).count()

System  Age  Occupation  \
Gender                            
Female      12   12          12   
Male        10   10          10   

        Highest level of education you have completed or currently pursuing  
Gender                                                                       
Female                                                 12                    
Male                                                   10

Most common level of education: Bachelor's degree

In [22]:
demographics[demographics.columns[-1]].mode()

0    Bachelor's degree
dtype: object

All levels of education

In [23]:
demographics[demographics.columns[-1]].value_counts()

Bachelor's degree       11
Doctoral degree          6
Master's degree          3
High school diploma      1
Post-doctoral degree     1
Name: Highest level of education you have completed or currently pursuing, dtype: int64

Most common occupation: student

In [24]:
demographics[demographics.columns[-2]].mode()

0    Student
dtype: object

All occupations

In [25]:
demographics[demographics.columns[-2]].str.lower().value_counts()

student                           5
graduate student                  2
front desk agent                  1
alumni relations manager (saa)    1
manager                           1
academic advisor                  1
director, it                      1
staff scientist                   1
writer                            1
software                          1
marketing manager                 1
dental hygienist                  1
editor                            1
phd student                       1
accounting                        1
pharmacist                        1
semi-retired                      1
Name: Occupation, dtype: int64

Mean age: 38.0

In [26]:
demographics[demographics.columns[-4]].mean()

38.0

Range of ages: (20, 60)

In [27]:
(demographics[demographics.columns[-4]].min(), demographics[demographics.columns[-4]].max())

(20, 60)

Grouping by system, age, and occupation

In [28]:
demographics.groupby(['System','Age','Occupation']).size()

System  Age  Occupation                    
Julie   20   Student                           1
        25   Student                           1
        26   Graduate Student                  1
             Student                           1
        29   PhD student                       1
        35   Director, IT                      1
        37   Staff scientist                   1
        49   Dental Hygienist                  1
        51   Front Desk Agent                  1
        54   manager                           1
        60   accounting                        1
Lily    26   Marketing Manager                 1
        27   Student                           1
        28   Graduate Student                  1
        33   Software                          1
        34   student                           1
        40   Academic advisor                  1
             Editor                            1
        41   Pharmacist                        1
        46   Alumni Relat

## Adjectives participants used to describe a self-driving vehicle

Here we try sentiment analysis to look at the adjectives participants provided to describe a self-driving vehicle

Checking the number of words that is in the opinion lexicon dataset
SOURCE: http://www.cs.uic.edu/~liub/FBS/sentiment-analysis.html

In [29]:
posArr = []
negArr = []
for word in open('positive-words.txt', 'r').readlines()[35:]:
    word = word.rstrip()
    posArr.append(word)
for word in open('negative-words.txt', 'r').readlines()[35:]:
    word = word.rstrip()
    negArr.append(word)
print ("num of positive words %d" % len(posArr))
print ("num of negative words %d" % len(negArr))

num of positive words 2006
num of negative words 4783


In [30]:
temp = df[df.columns[:-4]]
subj_words = temp.drop(temp.columns[1:-21],axis=1)

In [31]:
subj_words.shape

(22, 22)

First we create a dataframe that includes all questions of the form "Please choose the response that best represents how you perceive the self-driving vehicle..." (or similar) in which participants respond using a Likert scale or by filling in their own answer

In [32]:
subj_words.head()

System  \
0  Julie   
1   Lily   
2  Julie   
3  Julie   
4   Lily   

  Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.  \
0                          fun smart safe good enjoy                              
1    effortless, pleasant, robotic, time-saving, zen                              
2                                                NaN                              
3           smooth, safe, intuitive, automated, easy                              
4        safe, freedom, easy, self-sufficient, slow                               

  Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Capable]  \
0                                              Agree                                                                                                      
1                                            Neutral                                                                                                      
2                                              Agree                                                                                                      
3                                              Agree                                                                                                      
4                                              Agree                                                                                                      

  Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Emotional]  \
0                                           Disagree                                                                                                        
1                                  Strongly disagree                                                                                                        
2                                           Disagree                                                                                                        
3                                           Disagree                                                                                                        
4                                            Neutral                                                                                                        

  Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Energetic]  \
0                                              Agree                                                                                                        
1                                  Strongly disagree                                                                                                        
2                                           Disagree                                                                                                        
3                                            Neutral                                                                                                        
4                                            Neutral                                                                                                        

  Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Indifferent]  \
0                                           Disagree                                                                                                          
1                                     Strongly agree                                                                                                          
2                                              Agree                                                                                    

Creating a dataframe with responses from the question "Provide 5 words (adjectives) to characterize/describe a self-driving vehicle."

In [33]:
adjs = subj_words[subj_words.columns[:2]]

In [34]:
adjs.head()

System  \
0  Julie   
1   Lily   
2  Julie   
3  Julie   
4   Lily   

  Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.  
0                          fun smart safe good enjoy                             
1    effortless, pleasant, robotic, time-saving, zen                             
2                                                NaN                             
3           smooth, safe, intuitive, automated, easy                             
4        safe, freedom, easy, self-sufficient, slow

Removing NaN responses (namely, participants didn't answer this question). We see that 6 people didn't answer this question.

In [35]:
adjs2 = adjs.dropna()
print(len(adjs))
print(len(adjs)-len(adjs2)) # number of NaN, namely 6 people didn't provide adjs.

22
6


Dataframe without the NaN responses

In [36]:
adjs2

System  \
0   Julie   
1    Lily   
3   Julie   
4    Lily   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
11  Julie   
12   Lily   
13   Lily   
15  Julie   
16   Lily   
17  Julie   
19   Lily   
20  Julie   

   Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.  
0                           fun smart safe good enjoy                             
1     effortless, pleasant, robotic, time-saving, zen                             
3            smooth, safe, intuitive, automated, easy                             
4         safe, freedom, easy, self-sufficient, slow                              
6   advanced, futuristic, expensive, efficient, in...                             
7      convenient, smart, effective, time saving, new                             
8                    exciting hitech cool fun unknown                             
9   interesting, high-techy, scary, fascinating, s...                             
11  Trusting, high-tech, safer, blind-spot, futuri...                             
12                 simple beta future software assist                             
13         Futuristic, cautious, useful, boring, safe                             
15     useful, beneficial, environmentally sound, new                             
16  advanced, safe, technological, autonomy, reliable                             
17              Reliable, safe, easy, precise, useful                             
19  innovative, convenient, awesome, intelligent, ...                             
20  Logical, algorithmic, automated, safe, rule-ab...

In [37]:
adjs2.shape

(16, 2)

In [38]:
for index, row in adjs2.iterrows():
    if ',' in row[1]:
        row[1] = re.split(', ',row[1])
    else:
        row[1] = re.split(' ', row[1])

In [39]:
adjs2

System  \
0   Julie   
1    Lily   
3   Julie   
4    Lily   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
11  Julie   
12   Lily   
13   Lily   
15  Julie   
16   Lily   
17  Julie   
19   Lily   
20  Julie   

   Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.  
0                     [fun, smart, safe, good, enjoy]                             
1   [effortless, pleasant, robotic, time-saving, zen]                             
3          [smooth, safe, intuitive, automated, easy]                             
4       [safe, freedom, easy, self-sufficient, slow ]                             
6   [advanced, futuristic, expensive, efficient, i...                             
7    [convenient, smart, effective, time saving, new]                             
8              [exciting, hitech, cool, fun, unknown]                             
9   [interesting, high-techy, scary, fascinating, ...                             
11  [Trusting, high-tech, safer, blind-spot, futur...                             
12           [simple, beta, future, software, assist]                             
13       [Futuristic, cautious, useful, boring, safe]                             
15   [useful, beneficial, environmentally sound, new]                             
16  [advanced, safe, technological, autonomy, reli...                             
17            [Reliable, safe, easy, precise, useful]                             
19  [innovative, convenient, awesome, intelligent,...                             
20  [Logical, algorithmic, automated, safe, rule-a...

In [40]:
adjs_group_by_system = adjs2.groupby(['System'])
adjs_group_by_system.count()

Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.
System                                                                               
Julie                                                   8                            
Lily                                                    8

In [41]:
adjs2Temp = adjs2
for index, row in adjs2Temp.iterrows():
    posNum = 0
    negNum = 0
    for word in row[1]:
        word = word.lower()
        if word in posArr:
            posNum += 1
        if word in negArr:
            negNum -= 1
    total = posNum + negNum
    #print(posNum, negNum)
    adjs2.loc[index,'Total'] = total

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexing.py:362: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[key] = _infer_fill_value(value)
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pandas/core/indexing.py:543: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [42]:
adjs2

System  \
0   Julie   
1    Lily   
3   Julie   
4    Lily   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
11  Julie   
12   Lily   
13   Lily   
15  Julie   
16   Lily   
17  Julie   
19   Lily   
20  Julie   

   Provide 5 words (adjectives) to characterize/describe a self-driving vehicle.  \
0                     [fun, smart, safe, good, enjoy]                              
1   [effortless, pleasant, robotic, time-saving, zen]                              
3          [smooth, safe, intuitive, automated, easy]                              
4       [safe, freedom, easy, self-sufficient, slow ]                              
6   [advanced, futuristic, expensive, efficient, i...                              
7    [convenient, smart, effective, time saving, new]                              
8              [exciting, hitech, cool, fun, unknown]                              
9   [interesting, high-techy, scary, fascinating, ...                              
11  [Trusting, high-tech, safer, blind-spot, futur...                              
12           [simple, beta, future, software, assist]                              
13       [Futuristic, cautious, useful, boring, safe]                              
15   [useful, beneficial, environmentally sound, new]                              
16  [advanced, safe, technological, autonomy, reli...                              
17            [Reliable, safe, easy, precise, useful]                              
19  [innovative, convenient, awesome, intelligent,...                              
20  [Logical, algorithmic, automated, safe, rule-a...                              

    Total  
0     5.0  
1     2.0  
3     4.0  
4     4.0  
6     3.0  
7     3.0  
8     2.0  
9     2.0  
11    2.0  
12    0.0  
13    2.0  
15    2.0  
16    3.0  
17    5.0  
19    5.0  
20    2.0

In [43]:
(adjs2[adjs2.columns[2]].min(), adjs2[adjs2.columns[2]].max())

(0.0, 5.0)

In [44]:
adjs_group_by_system = adjs2.groupby(['System'])
adjs_group_by_system.sum()

Total
System       
Julie    25.0
Lily     21.0

Julie had slightly more positive words and an overall more positive sentiment

In [45]:
adjs_group_by_system.mean()

Total
System       
Julie   3.125
Lily    2.625

In [46]:
adjs_group_by_system_total = adjs2.groupby(['System','Total'])
adjs_group_by_system_total.size()

System  Total
Julie   2.0      4
        3.0      1
        4.0      1
        5.0      2
Lily    0.0      1
        2.0      3
        3.0      2
        4.0      1
        5.0      1
dtype: int64

## Mann-Whitney U Test: Sentiment of adjectives used to describe a self-driving vehicle

In [47]:
julie_adj = adjs2.loc[adjs2['System']=="Julie"][adjs2.columns[2]]
lily_adj = adjs2.loc[adjs2['System']=="Lily"][adjs2.columns[2]]

In [48]:
julie_adj.mean()

3.125

In [49]:
lily_adj.mean()

2.625

In [50]:
mannwhitneyu(julie_adj, lily_adj, alternative = 'two-sided')

MannwhitneyuResult(statistic=36.5, pvalue=0.6586874174078845)

Assuming $\alpha$ = 0.05, and given the resulting `pvalue = 0.6586874174078845` we see there does not seem to be a significant difference between the sentiment of words the two groups used to describe a self-driving vehicle.

## Questions where participants asked to rate words on Likert scale

We use the `subj_words` dataframe from earlier and exclude the question in which participants wrote in their own adjectives

In [51]:
choice_words = subj_words.drop(subj_words.columns[1],axis=1)

In [52]:
choice_words

System  \
0   Julie   
1    Lily   
2   Julie   
3   Julie   
4    Lily   
5   Julie   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
10   Lily   
11  Julie   
12   Lily   
13   Lily   
14   Lily   
15  Julie   
16   Lily   
17  Julie   
18  Julie   
19   Lily   
20  Julie   
21   Lily   

   Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Capable]  \
0                                               Agree                                                                                                      
1                                             Neutral                                                                                                      
2                                               Agree                                                                                                      
3                                               Agree                                                                                                      
4                                               Agree                                                                                                      
5                                               Agree                                                                                                      
6                                               Agree                                                                                                      
7                                               Agree                                                                                                      
8                                               Agree                                                                                                      
9                                             Neutral                                                                                                      
10                                     Strongly agree                                                                                                      
11                                              Agree                                                                                                      
12                                              Agree                                                                                                      
13                                              Agree                                                                                                      
14                                              Agree                                                                                                      
15                                              Agree                                                                                                      
16                                              Agree                                                                                                      
17                                              Agree                                                                                                      
18                                              Agree                                                                                                      
19                                              Agree                                                                                                      
20                                     Strongly agree                                                                                                      
21                                              Agree                                                                                                      

   Please choose the response that best represents how you perceive the self-driving vehicle. (The following words

In [53]:
choice_words.shape

(22, 21)

Replacing NaN with 0

In [54]:
choice_words = choice_words.replace({'Strongly disagree': 1,'Disagree': 2, 'Neutral': 3, 'Agree': 4, 'Strongly agree': 5})
choice_words = choice_words.fillna(0)
choice_words.groupby(['System']).mean()

Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Capable]  \
System                                                                                                                                                          
Julie                                                 4.0                                                                                                       
Lily                                                  4.0                                                                                                       

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Emotional]  \
System                                                                                                                                                            
Julie                                            2.090909                                                                                                         
Lily                                             2.000000                                                                                                         

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Energetic]  \
System                                                                                                                                                            
Julie                                            2.727273                                                                                                         
Lily                                             2.545455                                                                                                         

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Indifferent]  \
System                                                                                                                                                              
Julie                                            3.090909                                                                                                           
Lily                                             3.090909                                                                                                           

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Methodical]  \
System                                                                                                                                                             
Julie                                            3.636364                                                                                                          
Lily                                             3.909091                                                                                                          

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Predictable]  \
System                                                                                                                                                              
Julie                                            3.727273                                                                                                           
Lily                                             3.636364                                                                                                           

        Please choose the response that best represents how you perceive the s

In [55]:
choice_words.groupby(['System']).sum()

Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Capable]  \
System                                                                                                                                                          
Julie                                                  44                                                                                                       
Lily                                                   44                                                                                                       

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Emotional]  \
System                                                                                                                                                            
Julie                                                  23                                                                                                         
Lily                                                   22                                                                                                         

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Energetic]  \
System                                                                                                                                                            
Julie                                                  30                                                                                                         
Lily                                                   28                                                                                                         

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Indifferent]  \
System                                                                                                                                                              
Julie                                                  34                                                                                                           
Lily                                                   34                                                                                                           

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Methodical]  \
System                                                                                                                                                             
Julie                                                  40                                                                                                          
Lily                                                   43                                                                                                          

        Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Predictable]  \
System                                                                                                                                                              
Julie                                                  41                                                                                                           
Lily                                                   40                                                                                                           

        Please choose the response that best represents how you perceive the s

In [56]:
len(list(df)[13:33])

20

Participants answered 20 questions of the type "Please choose the response that best represents how you perceive the self-driving vehicle..."

## Mann Whitney U Test: perspectives on self-driving vehicles

In [57]:
df_choice_words = choice_words.assign(Choice_Words = choice_words[list(choice_words)[:]].sum(axis=1) / 20)

In [58]:
df_choice_words

System  \
0   Julie   
1    Lily   
2   Julie   
3   Julie   
4    Lily   
5   Julie   
6    Lily   
7   Julie   
8    Lily   
9   Julie   
10   Lily   
11  Julie   
12   Lily   
13   Lily   
14   Lily   
15  Julie   
16   Lily   
17  Julie   
18  Julie   
19   Lily   
20  Julie   
21   Lily   

    Please choose the response that best represents how you perceive the self-driving vehicle. (The following words are in alphabetical order.) [Capable]  \
0                                                   4                                                                                                       
1                                                   3                                                                                                       
2                                                   4                                                                                                       
3                                                   4                                                                                                       
4                                                   4                                                                                                       
5                                                   4                                                                                                       
6                                                   4                                                                                                       
7                                                   4                                                                                                       
8                                                   4                                                                                                       
9                                                   3                                                                                                       
10                                                  5                                                                                                       
11                                                  4                                                                                                       
12                                                  4                                                                                                       
13                                                  4                                                                                                       
14                                                  4                                                                                                       
15                                                  4                                                                                                       
16                                                  4                                                                                                       
17                                                  4                                                                                                       
18                                                  4                                                                                                       
19                                                  4                                                                                                       
20                                                  5                                                                                                       
21                                                  4                                                                                                       

    Please choose the response that best represents how you perceive the self-driving vehic

In [59]:
x = df_choice_words.loc[df_choice_words['System']=='Julie'][list(df_choice_words)[-1]]
y = df_choice_words.loc[df_choice_words['System']=='Lily'][list(df_choice_words)[-1]]
mannwhitneyu(x, y, alternative = 'two-sided')

MannwhitneyuResult(statistic=79.0, pvalue=0.23575734873508836)

In [60]:
x.mean(), y.mean()

(3.227272727272727, 2.75)

Not a statistically significant result.

Assuming $\alpha$ = 0.05, and given the resulting `pvalue = 0.23575734873508836` we see there does not seem to be a significant difference between the two groups' perspectives on self-driving vehicles.